EuclideanGeometry/Foundation/Axiom/Linear/Ray_ex2.lean

```diff
@@ -7,43 +7,232 @@ variable {P: Type _} [EuclideanPlane P]
 
+/- Given two segments $seg_1$ and $seg_2$, if the source and the target of the $seg_1$ both lie on $seg_2$, then every point of $seg_1$ lies on $seg_2$. -/
+theorem every_pt_lies_on_seg_of_source_and_target_lies_on_seg {seg₁ seg₂ : Seg P} (h₁ : seg₁.source LiesOn seg₂) (h₂ : seg₁.target LiesOn seg₂) {A : P} (ha : A LiesOn seg₁) : (A LiesOn seg₂) :=by
+  rcases h₁ with ⟨x,xnonneg,xle1,hx⟩
+  rcases h₂ with ⟨y,ynonneg,yle1,hy⟩
+  rcases ha with ⟨t,tnonneg,tleone,ht⟩
+  rw[←vec_add_vec seg₁.source seg₂.source,←vec_add_vec seg₁.source seg₂.source seg₁.target,←neg_vec,hx,hy,neg_add_eq_iff_eq_add,←neg_smul,smul_add,smul_smul,smul_smul,←add_smul,←add_smul,←add_assoc,mul_neg,←sub_eq_add_neg,←one_mul x,←mul_assoc,←sub_mul,mul_one] at ht
+  use ( (1- t) * x + t * y)
+  constructor
+  apply add_nonneg
+  apply mul_nonneg
+  linarith
+  linarith
+  apply mul_nonneg tnonneg ynonneg
+  constructor
+  nth_rw 2[←sub_add_cancel 1 t,←mul_one (1-t)]
+  nth_rw 4[←mul_one t]
+  apply add_le_add
+  apply mul_le_mul _ xle1 xnonneg
+  linarith
+  simp only [le_refl]
+  apply mul_le_mul _ yle1 ynonneg tnonneg
+  simp only [le_refl]
+  rw [ht]
+  
 /- Given two segments $seg_1$ and $seg_2$, if the source and the target of $seg_1$ both lie in the interior of $seg_2$, and if $A$ is a point on $seg_1$, then $A$ lies in the interior of $seg_2$. -/
-theorem
-every_pt_lies_int_seg_of_source_and_target_lies_int_seg {seg₁ seg₂ : Seg P} (h₂ : seg₁.source LiesInt seg₂) (h₂ : seg₁.target LiesInt seg₂) {A : P} (ha : A LiesOn seg₁) : A LiesInt seg₂ := sorry
+
+theorem every_pt_lies_int_seg_of_source_and_target_lies_int_seg {seg₁ seg₂ : Seg P} (h₁ : seg₁.source LiesInt seg₂) (h₂ : seg₁.target LiesInt seg₂) {A : P} (ha : A LiesOn seg₁) : A LiesInt seg₂ := by
+  rw[Seg.lies_int_iff]
+  constructor
+  apply ((Seg.lies_int_iff seg₂ seg₁.source).mp h₁).1
+  rw[Seg.lies_int_iff] at h₁ h₂
+  rcases h₁ with ⟨ _ ,x,xpos,xlt1,hx⟩
+  rcases h₂ with ⟨ _ ,y,ypos,ylt1,hy⟩
+  rcases ha with ⟨t,tnonneg,tle1,ht⟩
+  use ( (1- t) * x + t * y)
+  by_cases 0=t
+  constructor
+  simp only [←h, sub_zero, one_mul, zero_mul, add_zero]
+  exact xpos
+  constructor
+  simp only [←h, sub_zero, one_mul, zero_mul, add_zero]
+  exact xlt1
+  rw[←h,zero_smul,←eq_iff_vec_eq_zero] at ht
+  simp only [← h, sub_zero, one_mul, zero_mul, add_zero,ht,hx]
+  constructor
+  apply lt_of_lt_of_le (mul_pos (lt_of_le_of_ne tnonneg h) ypos)
+  simp only [le_add_iff_nonneg_left, gt_iff_lt, sub_pos]
+  apply mul_nonneg
+  linarith
+  linarith
+  constructor
+  have: (1-t)*x+t*y<(1-t)*x+t:=by
+    simp only [add_lt_add_iff_left, gt_iff_lt]
+    nth_rw 2[←mul_one t]
+    apply mul_lt_mul_of_pos_left ylt1 (lt_of_le_of_ne tnonneg h)
+  apply lt_of_lt_of_le this
+  have :1=1-t+t:=by norm_num
+  nth_rw 2 [this]
+  apply add_le_add
+  nth_rw 2[←mul_one (1-t)]
+  apply mul_le_mul
+  linarith
```

Comment:
it is possible to apply a single tactic to all current goals.
`tactic A <;> tactic B` or
```
tactic A <;>
tactic B
```
applys `tactic B` to all current goals.